usr/lib/common/mech_aes.c

```diff
@@ -488,6 +488,107 @@ CK_RV aes_ctr_decrypt(STDLL_TokData_t *tokdata,
     return rc;
 }
 
+CK_RV aes_xts_encrypt(STDLL_TokData_t *tokdata,
+                      SESSION *sess,
+                      CK_BBOOL length_only,
+                      ENCR_DECR_CONTEXT *ctx,
+                      CK_BYTE *in_data,
+                      CK_ULONG in_data_len,
+                      CK_BYTE *out_data, CK_ULONG *out_data_len)
+{
+	OBJECT *key = NULL;
+	CK_RV rc;
+
+	if (!sess || !ctx || !out_data_len) {
+		TRACE_ERROR("%s received bad argument(s)\n", __func__);
+		return CKR_FUNCTION_FAILED;
+	}
+	// CKM_DES3_CBC requires the input data to be an integral
+	// multiple of the block size
+	//
+	if (in_data_len % AES_BLOCK_SIZE != 0) {
```

Comment:
As far as I know for XTS the data must be at least a full block, but can be any number of bytes other than that. So for single part operation (this function), the check should be `if (in_data_len >= AES_BLOCK_SIZE) {`